src/aFIL.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "openzeppelin-contracts/token/ERC20/ERC20.sol";
+import "./LoanAgent.sol";
+
+error InsufficientTokenAvailable(uint availableAmount, uint requestAmount);
+
+contract aFIL is ERC20 {
+    uint public totalLoanAmount;
+    address public immutable oracle;
+
+    constructor(address _oracle) ERC20("arc FIL", "aFIL") {
+        oracle = _oracle;
+    }
+
+    struct Loan {
+        LoanAgent loanAgent;
+        uint totalAmount;
+        uint expectFinishTime;
+        uint remainingAmount;
+    }
+
+    mapping (address => Loan) loans;
+
+    modifier onlyOracle() {
```

Comment:
Oracle should not update the stat of aFIL directly. Instead LoanAgentFactory should have these privileges. replace `onlyOracle` with `onlyLoanAgentFactory`